internal/distributed/streamingnode/service.go

```diff
@@ -293,34 +302,46 @@ func (s *Server) initMeta() error {
 	return nil
 }
 
-func (s *Server) initRootCoord(ctx context.Context) (err error) {
-	log.Info("StreamingNode connect to rootCoord...")
-	s.rootCoord, err = rcc.NewClient(ctx)
-	if err != nil {
-		return errors.Wrap(err, "StreamingNode try to new RootCoord client failed")
-	}
+func (s *Server) initRootCoord(ctx context.Context) {
+	go func() {
+		retry.Do(ctx, func() error {
+			log.Info("StreamingNode connect to rootCoord...")
+			rootCoord, err := rcc.NewClient(ctx)
+			if err != nil {
+				return errors.Wrap(err, "StreamingNode try to new RootCoord client failed")
+			}
 
-	log.Info("StreamingNode try to wait for RootCoord ready")
-	err = componentutil.WaitForComponentHealthy(ctx, s.rootCoord, "RootCoord", 1000000, time.Millisecond*200)
-	if err != nil {
-		return errors.Wrap(err, "StreamingNode wait for RootCoord ready failed")
-	}
-	return nil
+			log.Info("StreamingNode try to wait for RootCoord ready")
+			err = componentutil.WaitForComponentHealthy(ctx, rootCoord, "RootCoord", 1000000, time.Millisecond*200)
+			if err != nil {
+				return errors.Wrap(err, "StreamingNode wait for RootCoord ready failed")
+			}
+			log.Info("StreamingNode wait for RootCoord done")
+			s.rootCoord.Set(rootCoord)
+			return nil
+		}, retry.AttemptAlways())
```

Comment:
Can we throw an error after the context ends instead of retrying indefinitely?

Reply:
retry will stop when error is context timeout or canceled